FSW/controllers/controller.py

```diff
@@ -0,0 +1,168 @@
+import configparser
+import numpy as np
+from world.math.quaternions import *
+from FSW.controllers.detumbling_controller import *
+
+class Controller:
+    def __init__(self, config, Magnetorquers, ReactionWheels, Idx) -> None:
+        self.config = config
+        self.pointing_mode    = config["pointing_mode"]
+        self.pointing_target  = config["pointing_target"]
+        self.controller_algo  = config["algorithm"]
+        self.tgt_ang_vel      = np.array(config["tgt_ss_ang_vel"], dtype=np.float64)
+        self.est_world_states = None
+        self.mtb              = Magnetorquers
+        self.G_mtb_b = np.array(config["mtb_orientation"]).reshape(config["N_mtb"], 3)
+        self.G_rw_b  = np.array(config["rw_orientation"]).reshape(config["N_rw"], 3)
+        self.inertia = np.array(config["inertia"]).reshape(3,3)
+        self.Bcrossctr  = BcrossController(self.G_mtb_b,
+                                            config["bcrossgain"],
+                                            Magnetorquers,
+                                            self.inertia,
+                                            config["tgt_ss_ang_vel"])
+        self.lyapsunpointctr = LyapBasedSunPointingController(self.inertia, 
+                                                              self.G_mtb_b,
+                                                              config["bcrossgain"],
+                                                              config["tgt_ss_ang_vel"],
+                                                              Magnetorquers)
+        self.basesunpointctr = BaselineSunPointingController(self.inertia, 
+                                                              self.G_mtb_b,
+                                                              config["bcrossgain"],
+                                                              np.array(config["mtb_att_feedback_gains"]),
+                                                              config["tgt_ss_ang_vel"],
+                                                              Magnetorquers)
+        self.nadirpointctr = BaselineNadirPointingController(self.inertia,
+                                                    self.G_mtb_b,
+                                                    config["bcrossgain"],
+                                                    np.array(config["mtb_att_feedback_gains"]),
+                                                    np.array(config["rw_att_feedback_gains"]),
+                                                    np.array(config["rw_vel_gain"]),
+                                                    np.array(config["tgt_ss_ang_vel"]),
+                                                    np.array(config["nom_rw_ang_vel"]),
+                                                    np.array(config["nadir_cam_dir"]),
+                                                    Magnetorquers,
+                                                    ReactionWheels)
```

Comment:
2 issues:
- I think we should only initialize one type of controller per generic controller instance. Ideally, every specialized controller inherits abstract methods from the base controller and adheres to some standard interfacing method dictated by the parent class.
- The constructor arguments are just not very readable. It would be better if there was standard object type being passed in (e.g. all numpy arrays), or if we pack all related data together neatly into a single class and pass that in (as is the case with the `Magnetorquers`).

Reply:
But these are just changes for long-term maintainability. If it's too much work, let's make a new branch for that after merging these changes

Reply:
Ive created a super class controller algorithm. We can edit further in the future